src/shims/io.rs

```diff
@@ -133,28 +154,53 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let count = this.read_scalar(count_op)?.to_usize(&*this.tcx)?;
```

Comment:
You need to delay the force ptr until it's actually needed, if at all

Reply:
If I delay it just before getting the allocation it also fails with the dangling pointer error. Also, if i try to just write to `&mut []` when `count == 0` it does not read any bytes at all.